lib/hammer_cli_foreman/defaults.rb

```diff
@@ -0,0 +1,27 @@
+require 'hammer_cli'
+module HammerCLIForeman
+
+  class Defaults < HammerCLI::Defaults
+    SUPPORTED_DEFAULTS=[:organization_id, :location_id]
+    def self.register_provider
+      super({self.name.to_sym => self})
+    end
+
+    def self.support?(param)
+      SUPPORTED_DEFAULTS.any? {|s| s.to_s == param}
+    end
+
+    def self.get_defaults(param)
+      param = "default_organization" if param == :organization_id
+      param = "default_location" if param == :location_id
+      user = HammerCLIForeman.foreman_resource(:users).action(:index).call(:search =>"login="+HammerCLI::Settings.settings[:foreman][:username])
+      user_id = user["results"][0]["id"] if user["results"]
+      values = HammerCLIForeman.foreman_resource(:users).action(:show).call(:id => user_id)
```

Comment:
Default organization and location are returned already in the index call so there's probably no need to do the second call here.